lib/srv/db/cloud/watchers/watcher.go

```diff
@@ -172,6 +194,55 @@ func makeFetchers(clients cloud.Clients, matchers []services.AWSMatcher) (result
 	return result, nil
 }
 
+func makeAzureFetchers(ctx context.Context, clients cloud.Clients, matchers []services.AzureMatcher) (result []Fetcher, err error) {
+	subIDsClient, err := clients.GetAzureSubscriptionIDsClient()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	for _, matcher := range matchers {
+		reduceAzureMatcher(&matcher)
+		subIDs := matcher.Subscriptions
+		if utils.SliceContainsStr(subIDs, types.Wildcard) {
+			// hit the subscriptions API at most once
+			subIDs, err = subIDsClient.ListSubscriptionIDs(ctx, common.MaxPages, true /*useCache*/)
+			if err != nil {
+				return nil, trace.Wrap(err)
+			}
```

Comment:
This code only runs at startup. I added it as a convenience for users only, but it would be weird UX to have `*` matching subscriptions but subscription changes are not picked up.

The simplest solution here is to list subscriptions every time `Fetcher.Get` is invoked on a fetcher configured with  `subscription: *`.

The other option is to remove `*` subscription matching, but I'm not in favor of that.

Reply:
sounds good, I'll make the change

Reply:
oh wait, if we wanted to fetch subIDs every time, then the `*` fetcher will be querying all subscriptions - whereas right now, I fetch all the subIDs and then make a fetcher for each one.

Reply:
I guess it doesn't really matter though... we call each fetcher sequentially anyway